odoo/addons/base/static/tests/base_settings_tests.js

```diff
@@ -178,5 +180,82 @@ QUnit.module('base_settings_tests', {
+                            '<div class="o_setting_search">' +
+                                '<input type="text" class="searchInput" placeholder="Search..."/>' +
+                                '<span class="searchIcon"><i class="fa fa-search" role="img" aria-label="Search" title="Search"/></span>' +
+                            '</div> ' +
+                        '</div> ' +
+                    '<header>' +
+                        '<button string="Save" type="object" name="execute" class="oe_highlight" />' +
+                        '<button string="Cancel" type="object" name="cancel" class="oe_link" />' +
+                    '</header>' +
+                    '</div>' +
+                    '<div class="o_setting_container">' +
+                        '<div class="settings_tab"/>'+
+                        '<div class="settings">' +
+                            '<div class="notFound o_hidden">No Record Found</div>' +
+                            '<div class="app_settings_block" string="CRM" data-key="crm">' +
+                                '<div class="o_setting_left_pane">' +
+                                    '<field name="foo"/>'+
+                                '</div>'+
+                                '<div class="o_setting_right_pane">'+
+                                    '<label for="foo"/>'+
+                                    '<div class="text-muted">'+
+                                        'this is foo'+
+                                    '</div>'+
+                                '</div>' +
+                            '</div>' +
+                            '<div class="app_settings_block" string="Other App" data-key="otherapp">' +
+                                '<div class="o_setting_left_pane">' +
+                                    '<field name="bar"/>'+
+                                '</div>'+
+                                '<div class="o_setting_right_pane">'+
+                                    '<label for="bar"/>'+
+                                    '<div class="text-muted">'+
+                                        'this is bar'+
+                                    '</div>'+
+                                '</div>' +
+                            '</div>' +
+                        '</div>' +
+                    '</div>' +
+                '</form>',
+        }).then(function (form) {
+            assert.notOk(form.$('.oe_form_configuration').is(':empty'),
+               "render should finish once deferred fields are resolved");
+            assert.ok(form.$('.tab.selected').is('[data-key="crm"]'));
+            assert.ok(form.$('.tab').is('[data-key="otherapp"]'));
+            assert.ok(form.$('.app_settings_block[data-key="crm"]').is(':visible'));
+            assert.ok(form.$('.app_settings_block[data-key="otherapp"]').is(':hidden'));
+            assert.ok($(document.activeElement).is('.searchInput'), "search has focus");
+            form.destroy();
```

Comment:
i'd do this in always handler as well

Reply:
if we have a `fail` we do not have `form` variable in `createAsyncView`.

so I think either we keep it as is, or put everything in `then` and expect createAsyncView to never fail

Reply:
haha you're right :-) sorry